### PR TITLE
resolve ansible-test issues

### PR DESCRIPTION
**Pull request by richm**
Lots of python code changes for formatting, fix linter issues,
etc.  This also cleans up the module docs.  One fix is that
for the `RETURN` parameters the `returned:` field is required.
I made all of the `returned: success` - I'm not sure if that
is really the case, or if some parameters are returned even if
the module failed, but I think `success` is safe.

### CODE REVIEW

**Comment by nhosoi** [comment]
LGTM

Looking at [this change](https://github.com/linux-system-roles/storage/pull/204/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L16-L21), is this the first time to run black and flake8 on the storage role? (no wonder there were lots of formatting changes... :)

**Comment by richm** [comment]
> LGTM
> 
> Looking at [this change](https://github.com/linux-system-roles/storage/pull/204/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L16-L21), is this the first time to run black and flake8 on the storage role? (no wonder there were lots of formatting changes... :)

Yes.  Since we have to make other big changes for ansible-test, I figured we should enable `black` and `flake8` too.

**Review by nhosoi — APPROVED** [review]
lgtm

(may not be merged just now, but you have my approval.)

**Comment by richm** [comment]
@nhosoi the 3 dependent PRs are merged - this PR is rebased on top of them - this PR also has the doc fixes

**Comment by richm** [comment]
rebase report - what affect with this have on existing PRs:
* #190 - minor change - https://github.com/richm/storage/commit/414db7258a13c782742fd208211cc1ea524408b9
* #189 - rebase is clean
* #187 - rebase is clean
* #185 - needs rebase to master first
* #181 - needs rebase to master first
* #169 - rebase is clean
* #150 - rebase is clean
* #149 - needs rebase to master first
* #121 - needs rebase to master first
* #104 - rebase is clean
* #94 - rebase is clean
* #50 - close in favor of #104?
* #47 - rebase is clean
* #39 - minor rebase conflict

So wdyt?  Can we merge this, then I help PR authors rebase their changes?

**Comment by richm** [comment]
ok to merge?

File: tests/unit/test_unused_disk.py [hunk 1]
```diff
@@ -1,72 +1,91 @@
...
-                     ('/dev/sdy', '/dev/sdy: PTTYPE=\"test\"')]
+blkid_data_pttype = [
+    ("/dev/sdx", '/dev/sdx: PTTYPE="dos"'),
+    ("/dev/sdy", '/dev/sdy: PTTYPE="test"'),
+]
 
-blkid_data = [('/dev/sdx', 'UUID=\"hello-1234-56789\" TYPE=\"crypto_LUKS\"'),
-              ('/dev/sdy', 'UUID=\"this-1s-a-t3st-f0r-ansible\" VERSION=\"LVM2 001\" TYPE=\"LVM2_member\" USAGE=\"raid\"'),
-              ('/dev/sdz', 'LABEL=\"/data\" UUID=\"a12bcdef-345g-67h8-90i1-234j56789k10\" VERSION=\"1.0\" TYPE=\"ext4\" USAGE=\"filesystem\"')]
+blkid_data = [
```
Comment (dwlehman) [inline tests/unit/test_unused_disk.py:15]: How this can be considered an improvement is beyond my comprehension, but I suppose I can read it either way.

**Review by dwlehman — APPROVED** [review]
A lot of these changes seem like bikeshedding to me, but I don't feel strongly about it.

File: tests/unit/test_unused_disk.py (hunk 1 above)
Comment (richm) [reply tests/unit/test_unused_disk.py:15]: python `black` - this is a double edged sword - it fixes many legitimate issues found by Galaxy importer, but it imposes a rather arbitrary, and in some cases hard-to-read, formatting regime

Comment (richm) [reply tests/unit/test_unused_disk.py:15]: Every other system role uses python `black` formatting - if storage doesn't want to use this, we'll have to figure out some way to make an exception, and figure out how to lint some of the issues found by Galaxy importer without using `black`

Comment (dwlehman) [reply tests/unit/test_unused_disk.py:15]: No, it's probably worth it for the consistency and linting.

Comment (richm) [reply tests/unit/test_unused_disk.py:15]: ok - here's https://github.com/linux-system-roles/storage/pull/205 for comparison - pretty much everything except the `black` formatting

**Comment by richm** [comment]
@dwlehman Can we get one of these PRs - this one or https://github.com/linux-system-roles/storage/pull/205 - merged soon?

**Comment by richm** [comment]
[citest pending]

**Comment by richm** [comment]
[citest bad]

**Comment by richm** [comment]
[citest pending]

**Comment by nhosoi** [comment]
@richm, Is https://github.com/linux-system-roles/storage/pull/205 a superset of this pr? Or independent?
Took a look quickly, some changes are in common, but others are not...
Probably, we need to merge both of them, but the second merge may give us some conflicts...

**Comment by richm** [comment]
> @richm, Is #205 a superset of this pr? Or independent?

Independent.  @dwlehman expressed concerns with the large number of formatting changes in this PR, so I created #205 as an alternative which fixed the ansible-lint issues without some of the large formatting changes, so we could compare the two and decide which one was better for the storage role.

> Took a look quickly, some changes are in common, but others are not...
> Probably, we need to merge both of them, but the second merge may give us some conflicts...

We need to choose one and close the other.

**Comment by dwlehman** [comment]
My comment was not intended to be a serious objection, just a comment that I personally find the new style unpleasant.

We should go with this version if it brings the project in line with the rest of the system roles in some meaningful way.

**Comment by richm** [comment]
closing in favor of https://github.com/linux-system-roles/storage/pull/205

**Pull request closed without merging**